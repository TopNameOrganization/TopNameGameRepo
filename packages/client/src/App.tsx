import { useEffect } from 'react'
import { AuthProvider } from './context/AuthContext'
import { Router } from './Router'
import { QueryClientProvider, QueryClient } from 'react-query'
import { ErrorBoundary } from './components/ErrorBoundary'
import { DefaultStub } from './components/DefaultStub'
import { serviceWorkersRegistration } from './serviceWorkersRegistration'
import { CustomThemeProvider } from './context/ThemeProvider'
import React from 'react'

const queryClient = new QueryClient()

function App() {
  useEffect(() => {
    serviceWorkersRegistration()
  }, [])

  return (
    <QueryClientProvider client={queryClient}>
      <ErrorBoundary msg={<DefaultStub />}>
        <AuthProvider>
          <CustomThemeProvider>
            <Router />
          </CustomThemeProvider>
        </AuthProvider>
      </ErrorBoundary>
    </QueryClientProvider>
  )
}

export default App
