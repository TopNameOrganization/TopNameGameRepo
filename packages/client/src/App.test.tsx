import App from './App'
import * as renderer from "react-test-renderer";
import { BrowserRouter } from 'react-router-dom'

test('should renders App correctly', async () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
})
