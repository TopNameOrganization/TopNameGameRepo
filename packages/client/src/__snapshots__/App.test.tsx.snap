// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders App correctly 1`] = `
<div
  className="MuiBox-root css-ue6sde"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthMd css-xdmu94-MuiContainer-root"
  >
    <h1
      className="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
    >
      Error
    </h1>
    <h6
      className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
    >
      Oops, something went wrong
    </h6>
    <a
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-19fqo1i-MuiButtonBase-root-MuiButton-root"
      href="#/"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
    >
      Back Home
    </a>
  </div>
</div>
`;
